cmd/baton-confluence-datacenter/main.go

```diff
@@ -5,46 +5,67 @@ import (
 	"fmt"
 	"os"
 
-	"github.com/conductorone/baton-sdk/pkg/cli"
+	"github.com/conductorone/baton-confluence-datacenter/pkg/connector"
+	configSchema "github.com/conductorone/baton-sdk/pkg/config"
 	"github.com/conductorone/baton-sdk/pkg/connectorbuilder"
+	"github.com/conductorone/baton-sdk/pkg/field"
 	"github.com/conductorone/baton-sdk/pkg/types"
 	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"github.com/spf13/viper"
 	"go.uber.org/zap"
+)
 
-	"github.com/conductorone/baton-confluence-datacenter/pkg/connector"
+const (
+	version       = "dev"
+	connectorName = "baton-confluence-datacenter"
+	accessToken   = "access-token"
+	hostname      = "hostname"
+	password      = "password"
+	username      = "username"
 )
 
-var version = "dev"
+var (
+	accessTokenField    = field.StringField(accessToken, field.WithDescription("The personal access token for your Confluence Data Center account."))
+	hostnameField       = field.StringField(hostname, field.WithRequired(true), field.WithDescription("The hostname (URL) for your Confluence Data Center."))
+	passwordField       = field.StringField(password, field.WithDescription("The password for your Confluence Data Center account."))
+	usernameField       = field.StringField(username, field.WithDescription("The username for your Confluence Data Center account."))
+	configurationFields = []field.SchemaField{accessTokenField, hostnameField, passwordField, usernameField}
+	fieldRelationships  = []field.SchemaFieldRelationship{
+		field.FieldsAtLeastOneUsed(accessTokenField, usernameField),
+		field.FieldsDependentOn(
```

Comment:
This should be 
```go
field.FieldsRequiredTogether(
  usernameField, 
  passwordField,
),
```
otherwise the user would be able to pass a password without a username.

Reply:
I think it would also be helpful to add a comment on the relationships list saying that username/password and access token aren't explicitly mutually exclusive, but that only one set of credentials are used.

Alternatively, we can just make them mutually exclusive.

Reply:
It is weird, I am not able to pass a password without a username
./dist/darwin_arm64/baton-confluence-datacenter --hostname http://localhost:3000 --password mysecretpwd
at least one of the flags in the group [access-token username] is required

Reply:
Fixed